trunk/source/Library-UltimateModel/src/de/uni_freiburg/informatik/ultimate/core/model/models/annotation/IMessageProvider.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2023 Manuel Bentele
+ *
+ * This file is part of the ULTIMATE Core.
+ *
+ * The ULTIMATE Core is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ULTIMATE Core is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ULTIMATE Core. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Additional permission under GNU GPL version 3 section 7:
+ * If you modify the ULTIMATE Core, or any covered work, by linking
+ * or combining it with Eclipse RCP (or a modified version of Eclipse RCP),
+ * containing parts covered by the terms of the Eclipse Public License, the
+ * licensors of the ULTIMATE Core grant you additional permission
+ * to convey the resulting work.
+ */
+
+package de.uni_freiburg.informatik.ultimate.core.model.models.annotation;
+
+import java.util.function.Supplier;
+
+/**
+ * Message provider interface returning messages for {@link ISpec.Type}s.
+ * 
+ * @author Manuel Bentele
+ */
+public interface IMessageProvider {
+
+	/**
+	 * Returns a positive default message for a given {@link ISpec.Type}.
+	 * 
+	 * @param spec
+	 *            a specification that is checked.
+	 * 
+	 * @return positive default message for the given {@code spec}.
+	 */
+	public default String getDefaultPositiveMessage(final ISpec.Type spec) {
```

Comment:
public is not necessary in a public interface

Reply:
But it's not forbidden either ;-) Jokes aside, I will remove it.